core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -0,0 +1,304 @@
+		return err
+	}
+	//display the requested asset agreement
+	log.Info(fmt.Printf("assetExchangeAgreement: %+v\n", assetAgreement))
+
+	lockInfoHTLC := &common.AssetLockHTLC{}
+	err = proto.Unmarshal([]byte(lockInfoBytes), lockInfoHTLC)
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+	//display the requested asset agreement
+	log.Info(fmt.Printf("lockInfoHTLC: %+v\n", lockInfoHTLC))
+
+	if lockInfoHTLC.TimeSpec != common.AssetLockHTLC_EPOCH {
+		errorMsg := "only EPOCH time is supported at present"
+		log.Error(errorMsg)
+		return errors.New(errorMsg)
+	}
+
+	assetLockKey := assetAgreement.Type + ":" + assetAgreement.Id
+	assetLockVal := AssetLockValue{Locker: assetAgreement.Locker, Recipient: assetAgreement.Recipient, Hash: string(lockInfoHTLC.Hash), ExpiryTimeSecs: lockInfoHTLC.ExpiryTimeSecs}
+
+	assetLockValBytes, err := ctx.GetStub().GetState(assetLockKey)
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+
+        if assetLockValBytes != nil {
+		errorMsg := fmt.Sprintf("Asset of type %s and ID %s is already locked", assetAgreement.Type, assetAgreement.Id)
+		log.Error(errorMsg)
+		return errors.New(errorMsg)
+	}
+
+	assetLockValBytes, err = json.Marshal(assetLockVal)
+	if err != nil {
+		errorMsg := fmt.Sprintf("Marshal error: %s", err)
+		log.Error(errorMsg)
+		return errors.New(errorMsg)
+	}
+	return ctx.GetStub().PutState(assetLockKey, assetLockValBytes)
+}
+
+// UnLockAsset cc is used to record unlocking of an asset on the ledger
+func (s *SmartContract) UnLockAsset(ctx contractapi.TransactionContextInterface, assetAgreementBytes string) error {
+
+	assetAgreement := &common.AssetExchangeAgreement{}
+	err := proto.Unmarshal([]byte(assetAgreementBytes), assetAgreement)
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+	//display the requested asset agreement
+	log.Info(fmt.Printf("assetExchangeAgreement: %+v\n", assetAgreement))
+
+	assetLockKey := assetAgreement.Type + ":" + assetAgreement.Id
+
+	assetLockValBytes, err := ctx.GetStub().GetState(assetLockKey)
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+
+        if assetLockValBytes == nil {
+		errorMsg := fmt.Sprintf("No asset of type %s and ID %s is locked", assetAgreement.Type, assetAgreement.Id)
+		log.Error(errorMsg)
+		return errors.New(errorMsg)
+	}
+
+	assetLockVal := AssetLockValue{}
+	err = json.Unmarshal(assetLockValBytes, &assetLockVal)
+	if err != nil {
+		errorMsg := fmt.Sprintf("Unmarshal error: %s", err)
+		log.Error(errorMsg)
+		return errors.New(errorMsg)
+	}
+
+	if assetLockVal.Locker != assetAgreement.Locker || assetLockVal.Recipient != assetAgreement.Recipient {
+		errorMsg := fmt.Sprintf("Cannot unlock asset of type %s and ID %s as it is locked by %s for %s", assetAgreement.Type, assetAgreement.Id, assetLockVal.Locker, assetLockVal.Recipient)
+		log.Error(errorMsg)
+		return errors.New(errorMsg)
+	}
+
+	// Check if expiry time is elapsed
+	currentTimeSecs := time.Now().UnixNano() / int64(time.Second)
```

Comment:
is there a reason we can't just use `time.Now().Unix()` instead? https://golang.org/pkg/time/#Time.Unix
_comment applies to the 5 other similar instances_

Reply:
Done.